install/kubernetes/cilium/templates/cilium-agent-daemonset.yaml

```diff
@@ -457,6 +486,24 @@ spec:
           path: /sys/fs/bpf
           type: DirectoryOrCreate
         name: bpf-maps
+{{- end }}
+{{- if .Values.cgroupAutoMount.enabled }}
+      # To mount cgroup filesystem on the host
+      - hostPath:
+          path: /proc/1/ns
```

Comment:
Could we limit the mount scope even more? I.e. to mount only `/proc/1/ns/cgroup` and `/proc/1/ns/mnt`?

Reply:
Those are just symlinks.

Reply:
From namespaces(7):
>        Bind mounting (see mount(2)) one of the files in this directory
>       to somewhere else in the filesystem keeps the corresponding
>       namespace of the process specified by pid alive even if all
>       processes currently in the namespace terminate.
To my understanding, mounting only those files should work.

Reply:
@brb I'm not sure that's supposed to work. This might be relate d- https://stackoverflow.com/questions/34783391/why-i-couldnt-use-mount-bind-proc-pid-ns-mnt-to-another-file-in-ubuntu

`  Warning  Failed     20s                kubelet            Error: failed to start container "mount-cgroup": Error response from daemon: OCI runtime create failed: container_linux.go:349: starting container process caused "process_linux.go:449: container init caused \"rootfs_linux.go:58: mounting \\\"/proc/1/ns/mnt\\\" to rootfs \\\"/var/lib/docker/overlay2/7b5659481002373db731ac03d0563a9cad37be0027d77bf755a8332a32c080e4/merged\\\" at \\\"/var/lib/docker/overlay2/7b5659481002373db731ac03d0563a9cad37be0027d77bf755a8332a32c080e4/merged/hostpid1mntns\\\" caused \\\"invalid argument\\\"\"": unknown`

Reply:
@aditighag try changing the type from `DirectoryOrCreate` to `Directory`

Reply:
@aanm I'm not sure which line you were pointing to. Anyway I meant to fix the type for `/proc/1/ns mount`. The DS shouldn't try to create that directory. 😅 ✅